mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -243,6 +245,19 @@ class ChiSqSelector @Since("2.1.0") () extends Serializable {
       case ChiSqSelector.FPR =>
         chiSqTestResult
           .filter { case (res, _) => res.pValue < alpha }
+      case ChiSqSelector.FDR =>
+        val tempRDD = chiSqTestResult
+          .sortBy { case (res, _) => res.pValue }
+        val maxIndex = tempRDD
+          .zipWithIndex
```

Comment:
I have added a large size data sample in the test Suite, and updated the Contingency tables in the test Suite comments. The value of degree of freedom, statistic and pValue for each feature is also added. so it is easy to validate the result.

SelectFDR in sklearn is not exact. My PR to fix the bug is merged today.